### PR TITLE
More cleaning in statements.sty.ltxml

### PR DESCRIPTION
**Pull request by angerhang**
Based on Deyan's advice cleaner `optionHelper` sub is made,

in addition to a quick patch to the `defindex` option handling.

### CODE REVIEW

File: sty/statements/statements.dtx [hunk 1]
```diff
@@ -1511,22 +1511,17 @@ sub defHelper{
...
-     $name = $whatsit->getArg(2)->toString.'-'.$whatsit->getArg(3)->toString unless $name;
-  } elsif($defOption eq 'adefii') {
-     $name = $whatsit->getArg(3)->toString.'-'.$whatsit->getArg(4)->toString unless $name;
-  } elsif($defOption eq 'defiii') {
-     $name = $whatsit->getArg(2)->toString.'-'.$whatsit->getArg(3)->toString.'-'.$whatsit->getArg(4)->toString unless $name;
-  } elsif($defOption eq 'adefiii') {
-     $name = $whatsit->getArg(3)->toString.'-'.$whatsit->getArg(4)->toString.'-'.$whatsit->getArg(5)->toString unless $name;
-   }
+  my %choose_Option = (
+                      definiendum => sub {$whatsit->setProperty(name=>$name->toString);
```
Comment (dginev) [inline sty/statements/statements.dtx]: The ```setProperty``` doesn't really belong there, you should move it out of the function and after ```$name``` is assigned to. It never hurts to have the name property set, so you can just add it for all cases.

Comment (dginev) [inline sty/statements/statements.dtx]: Looks much cleaner though, nice refactoring!

Comment (angerhang) [inline sty/statements/statements.dtx]: Thanks!

Just did the change and it should be ready to merge now : )

**Comment by dginev** [comment]
Nice work here, happy to see this tidy bit merged.

**Pull request merged**